podcasts/Sharing/Clip/VideoExporter.swift

```diff
@@ -0,0 +1,203 @@
+                videoWriter.finishWriting {
+                    continuation.resume()
+                }
+            }
+        }
+    }
+
+    @available(iOS 16, *)
+    @MainActor
+    private static func pixelBuffer(for view: some View, size: CGSize) throws -> UnsafeTransfer<CVPixelBuffer> {
+        try UnsafeTransfer(view.frame(width: size.width, height: size.height).pixelBuffer(size: size))
+    }
+
+    // Part 2 of video export, creating the final track from the initial video loop
+    private static func createFinalComposition(from sourceURL: URL, with parameters: Parameters, to outputURL: URL, progress: Progress) async throws -> AVComposition {
+        let asset = AVAsset(url: sourceURL)
+        let composition = AVMutableComposition()
+
+        guard let compositionVideoTrack = composition.addMutableTrack(withMediaType: .video, preferredTrackID: kCMPersistentTrackID_Invalid),
+              let sourceVideoTrack = try? await asset.loadTracks(withMediaType: .video).first else {
+            throw ExportError.failedToCreateCompositionTrack
+        }
+
+        let sourceTimeRange = try await asset.load(.duration)
+        var currentTime: CMTime = .zero
+
+        while currentTime < CMTime(seconds: parameters.duration, preferredTimescale: 600) {
+            try compositionVideoTrack.insertTimeRange(CMTimeRange(start: .zero, duration: sourceTimeRange),
+                                                      of: sourceVideoTrack,
+                                                      at: currentTime)
+            currentTime = CMTimeAdd(currentTime, sourceTimeRange)
+        }
+
+        if let audioTrack = try await parameters.audioPlayerItem.asset.loadTracks(withMediaType: .audio).first {
+            try add(audioTrack: audioTrack, at: CMTimeRange(start: parameters.audioStartTime, duration: parameters.audioDuration), to: composition)
+        }
+
+        return composition
+    }
+
+    // Part of Step 2 of video export to export the final file
+    private static func exportFinalComposition(_ composition: AVComposition, to outputURL: URL, duration: Double, fileType: AVFileType, progress: Progress) async throws {
+        guard let exportSession = AVAssetExportSession(asset: composition, presetName: AVAssetExportPresetHighestQuality) else {
+            throw ExportError.failedToCreateExportSession
+        }
+
+        exportSession.outputURL = outputURL
+        exportSession.outputFileType = fileType
+        exportSession.timeRange = CMTimeRange(start: .zero, duration: CMTime(seconds: duration, preferredTimescale: 600))
+
+        await exportSession.export()
+
+        guard exportSession.status == .completed else {
+            throw ExportError.exportFailed(exportSession.error)
+        }
+    }
+
+    // Part of Step 2 of video export to add the audio track
+    private static func add(audioTrack: AVAssetTrack, at timeRange: CMTimeRange, to composition: AVMutableComposition) throws {
```

Comment:
Just a helper which throws a `failedToAddAudioTrack` error if `addMutableTrack` failes for some reason